examples/android/app/src/main/java/com/cobaltspeech/cubicdemo/MainActivity.java

```diff
@@ -0,0 +1,370 @@
+// Copyright (2019) Cobalt Speech and Language, Inc. All rights reserved.
+
+package com.cobaltspeech.cubicdemo;
+
+import android.annotation.SuppressLint;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.Switch;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.cobaltspeech.cubic.CubicGrpc;
+import com.cobaltspeech.cubic.CubicOuterClass;
+import com.cobaltspeech.cubic.CubicOuterClass.ListModelsRequest;
+import com.cobaltspeech.cubic.CubicOuterClass.ListModelsResponse;
+import com.cobaltspeech.cubic.CubicOuterClass.Model;
+import com.cobaltspeech.cubic.CubicOuterClass.RecognitionAudio;
+import com.cobaltspeech.cubic.CubicOuterClass.RecognitionConfig;
+import com.cobaltspeech.cubic.CubicOuterClass.RecognitionResponse;
+import com.cobaltspeech.cubic.CubicOuterClass.StreamingRecognizeRequest;
+import com.cobaltspeech.cubic.CubicOuterClass.VersionResponse;
+import com.google.protobuf.ByteString;
+
+import java.util.ArrayList;
+import java.util.Locale;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+
+
+public class MainActivity extends AppCompatActivity {
+    private static final String TAG = MainActivity.class.getSimpleName();
+    private static final int SAMPLE_BLOCK_SIZE = 8192;
+
+    StringBuilder mTranscriptionText; // Holder for text transcripts to be displayed
+
+    private EditText editText_URL;
+    private Switch switchSecureConnection;
+    private Button buttonConnect;
+    private Spinner spinnerModels;
+    private Button buttonPushToTalk;
+    private TextView textViewResults;
+    private TextView labelVersions;
+    private AudioRecorderWrapper mAudioRecorder;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
```

Comment:
I should probably know this, but what is `R`? Is that a resource bundle?

Reply:
Yes.